src/server/protocol.ts

```diff
@@ -1707,15 +1709,13 @@ namespace ts.server.protocol {
          */
         prefix?: string;
         /**
-         * If enabled, TypeScript will search through all external modules' exports and add them to the completions list.
-         * This affects lone identifier completions but not completions on the right hand side of `obj.`.
+         * @deprecated Use Options
          */
-        includeExternalModuleExports: boolean;
+        includeExternalModuleExports?: boolean;

```

Comment:
not sure why should these be lumped with the other... i mean we can change to a global config system, but maybe on a separate change, and then we need to discuss that as well with the VSCode folks and with @amcasey

Reply:
Spoke to @amcasey and seems like it will work for him. so i am fine with he change. we probably need to give them more descriptive names.

Reply:
Wouldn't changing the name be a breaking change though? Although we could deprecate the existing name and add an equivalent one.

Reply:
I think @mhegazy meant that we should give them more descriptive names in their new location.  In their old location, they have to stay as-is for back-compat.

Reply:
@amcasey Do you have some good suggestions for the new names?

Reply:
I had envisioned that they would be in a nested object called (e.g.) "completions".  However, if that's not idiomatic, incorporating "completions" into the name should be sufficient - e.g. "includeCompletionsForExternalModuleExports".